replication.tf

```
@@ -1,7 +1,7 @@
 resource "aws_iam_role" "replication" {
   count = local.replication_enabled ? 1 : 0
 
-  name                 = format("%s-replication", module.this.id)
+  name                 = "${aws_s3_bucket.default[0].id}-replication"
```

Wouldn't this break existing implementations that use the module.this.id contextual inputs ?

---

We may want to consider this instead

```suggestion
  name                 = format("%s-replication, local.bucket_name)
```